crates/comm/src/socket/packet.rs

```diff
@@ -104,6 +108,8 @@ impl From<u8> for SocketPacketType {
             2 => SocketPacketType::SynAck,
             3 => SocketPacketType::Heartbeat,
             4 => SocketPacketType::Data,
+            5 => SocketPacketType::RequestAvailablePeers,
+            6 => SocketPacketType::SendAvailablePeers,
```

Comment:
I'm still very unhappy that this is implemented at the socket level. If you use the existing data layer, it will handle retransmission and everything for you.